internal/services/compute/image_data_source.go

```diff
@@ -0,0 +1,60 @@
+package compute
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/compute/mgmt/compute"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurestack/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurestack/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurestack/internal/tf/timeouts"
+)
+
+func ImageDataSource() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: imageDataSourceRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+
+			"resource_group_name": commonschema.ResourceGroupName(),
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
```

Comment:
should we be exporting other properties here?

Reply:
Hi @katbyte ,
Thanks for your advice. I used https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/compute/image_data_source.go for reference. It is good to have in Azurestack. And it can find the image by using the name and the resourcegroup.

Reply:
that one has a bunch of additional properties, could we get the ones supported added here? thanks!

Reply:
@katbyte 
documentation added
test updated
more attributes exposed